# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/nsNativeDragTarget.cpp
# Commit: 1fa3e699ce84
# Full Hash: 1fa3e699ce84a2c7562e25e2e3a6c1194372042f
# Author: David Parks <davidp99@gmail.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 2 - Make C++ callers of nsIDragService::GetCurrentSession pass a widget r=gstoll,win-reviewers,geckoview-reviewers,m_kato
#   
#   Content process callers must call nsIDragService::GetCurrentSession with the
#   correct widget under the drag.  For all callers, it is a sanity check.
#   
# ==============================================================================

diff -r 21e7f9ac67f8 -r 1fa3e699ce84 widget/windows/nsNativeDragTarget.cpp
--- a/widget/windows/nsNativeDragTarget.cpp	Thu Jul 04 07:48:03 2024 +0000
+++ b/widget/windows/nsNativeDragTarget.cpp	Thu Jul 04 07:48:03 2024 +0000
@@ -166,8 +166,8 @@
   GetGeckoDragAction(grfKeyState, pdwEffect, &geckoAction);
 
   // Set the current action into the Gecko specific type
-  nsCOMPtr<nsIDragSession> currSession;
-  mDragService->GetCurrentSession(getter_AddRefs(currSession));
+  RefPtr<nsDragSession> currSession =
+      static_cast<nsDragSession*>(mDragService->GetCurrentSession(mWidget));
   if (!currSession) {
     return;
   }
@@ -298,8 +298,8 @@
   // then we should include it as an allowed effect.
   mEffectsAllowed = (*pdwEffect) | (mEffectsAllowed & DROPEFFECT_LINK);
 
-  nsCOMPtr<nsIDragSession> currentDragSession;
-  mDragService->GetCurrentSession(getter_AddRefs(currentDragSession));
+  nsCOMPtr<nsIDragSession> currentDragSession =
+      mDragService->GetCurrentSession(mWidget);
   if (!currentDragSession) {
     return S_OK;  // Drag was canceled.
   }
@@ -349,8 +349,8 @@
   // dispatch the event into Gecko
   DispatchDragDropEvent(eDragExit, gDragLastPoint);
 
-  nsCOMPtr<nsIDragSession> currentDragSession;
-  mDragService->GetCurrentSession(getter_AddRefs(currentDragSession));
+  nsCOMPtr<nsIDragSession> currentDragSession =
+      mDragService->GetCurrentSession(mWidget);
 
   if (currentDragSession) {
     nsCOMPtr<nsINode> sourceNode;
@@ -426,8 +426,8 @@
   // Now process the native drag state and then dispatch the event
   ProcessDrag(eDrop, grfKeyState, aPT, pdwEffect);
 
-  nsCOMPtr<nsIDragSession> currentDragSession;
-  serv->GetCurrentSession(getter_AddRefs(currentDragSession));
+  nsCOMPtr<nsIDragSession> currentDragSession =
+      serv->GetCurrentSession(mWidget);
   if (!currentDragSession) {
     return S_OK;  // DragCancel() was called.
   }
